src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,194 @@
+    this.emailChangeHandler = this.emailChangeHandler.bind(this);
+  }
+
+  async OKHandler() {
+    const ticket = {};
+    ticket.request = {
+      requester: { name: this.state.name, email: this.state.email },
+      subject: this.state.subject,
+      // BEWARE changing the following ids or values! If you change them then you must thoroughly test.
+      custom_fields: [
+        { id: 360012744452, value: this.state.type},
+        { id: 360007369412, value: this.state.description},
+        { id: 360012744292, value: this.state.name},
+        { id: 360012782111, value: this.state.email },
+        { id: 360018545031, value: this.state.email }
+      ],
+      comment: {
+        body: this.state.description + '\n\n------------------\nSubmitted from: ' + this.props.url,
+        //${currUrl}
+        uploads: [this.state.attachment.token]
+      },
+      ticket_form_id: 360000669472
+    };
+    await Support.createSupportRequest(ticket);
+    
+    await this.setState(prev => {
+      prev.name = '';
+      prev.type = "question";
+      prev.subject = '';
+      prev.description = '';
+      prev.attachment = '';
+      prev.email = '';
+      return prev;
+    });
+    this.props.onOKRequest('support');;
+  }
+
+  closeHandler() {
+    // this is the method to handle Cancel click
+    // could do some cleaning here
+    // or delegate it to the parent
+    // we need to use it to close the
+    // DO SOMETHING HERE ...
+
+    // and call parent's close handler
+    this.props.onCloseRequest('support');
+  }
+
+  afterOpenHandler() {
+    // call parent's after open handler
+    this.props.onAfterOpen('support');
+
+  }
+
+  nameChangeHandler = (e) => {
+    const nameText = e.target.value;
+    this.setState(prev => {
+      prev.name = nameText;
+      return prev;
+    });
+  };
+
+  typeChangeHandler = (e) => {
+    const typeText = e.target.value;
+    this.setState(prev => {
+      prev.type = typeText;
+      return prev;
+    });
+  };
+
+  subjectChangeHandler = (e) => {
+    const subjectText = e.target.value;
+    this.setState(prev => {
+      prev.subject = subjectText;
+      return prev;
+    });
+  };
+
+  descriptionChangeHandler = (e) => {
+    const descriptionText = e.target.value;
+    this.setState(prev => {
+      prev.description = descriptionText;
+      return prev;
+    });
+  };
+
+  async attachmentChangeHandler(e){
+    const file = e.target.files[0];
+    const attachmentText = await Support.uploadAttachment(file);
+    this.setState(prev => {
+      prev.attachment = attachmentText;
+      return prev;
+    });
+  };
+
+  emailChangeHandler = (e) => {
+    let emailText = e.target.value;
+    this.state.vaild = /.+@.+\.[A-Za-z]+$/.test(emailText) ? true: false; 
+    this.setState(prev => {
+      prev.email = emailText;
+      return prev;
+    });
+  };
+
+  render() {
+
+    return (
+      SupportRequestBaseModal({
+        id: "SupportRequestModal",
```

Comment:
Here and throughout, use single quotes where possible.
```suggestion
        id: 'SupportRequestModal',
```